### ISSUE TITLE
STLs with ASCII description and binary overrunning binary data into description

### ISSUE THREAD

**Issue by stuartpb** [tmpvar/stl#2]
Test case: Download [this file](https://www.thingiverse.com/download:2672487) and parse it with `stl.toObject`.

Expected behavior: The object description should be "BRH5-GAUNTLET-LITE", and the rest of the object should contain all the facets defined in the file.

Observed behavior: The object description contains the entire first chunk read.

**Comment by stuartpb** [comment]
Whoa, actually, looking at this file a little closer in a hex editor, apparently it *does* have ASCII content, but, like, after a big binary chunk. WTF?

**Comment by stuartpb** [comment]
Okay, my initial read of the issue misstated the problem.

I think the bug lies [here](https://github.com/tmpvar/stl/blob/master/stl.js#L226): when the file is being read in as binary, the description should be truncated at the first null.

**Comment by tmpvar** [comment]
Hey, sorry it took so long to get my eyes on this. This is basically the dark corner in the STL "spec" and its resolution is ambiguous. 

From wikipedia:
> A binary STL file has an 80-character header (which is generally ignored, but should never begin with "solid" because that may lead some software to assume that this is an ASCII STL file).

Which is exactly what we are seeing here. Due to the streaming nature of this library, I'm not sure there is a good way to approach this. Here are the constraints that we have to deal with:

- __binary__: 80 byte 'description' header, which can be completely filled (e.g., no `\0` bytes)
- __ascii__: The first line is `solid <description of arbitrary length>`.

So the question is, when do we know that the stl is binary vs ascii when the first bytes are `solid`?

---

I think the best course of action here would be to read ahead and make the ascii/binary determination later on when a "facet"/"endfacet" token is reached.

**Comment by stuartpb** [comment]
Yeah, I mean, the way I was thinking you'd do it is you'd just read the first 81 bytes before emitting anything, and if the 81st character starts a binary STL, that's your clue that the first 80 characters were an ASCII header - it sounds like that's more or less what #4 is set to do.
